### ISSUE TITLE
failed to connect to taos using python client

### ISSUE THREAD

**Issue by zhanghongquan** [taosdata/TDengine#5985]
**Bug Description**
failed to connect to taos server using python client, got exception as
'taos.error.ProgrammingError: Unable to establish connection'

**To Reproduce**
1. setup tdengine server (2.0.18.0)
2.  install python client library(2.0.18.0)
3. execute read_example.py coming from tdengine's source package

**Expected Behavior**
run read_example.py successfully

**Environment (please complete the following information):**
 - OS: [Ubuntu 20.04]
 - TDengine Version [2.0.18.0]

**Additional Context**
server port started from 26030 as configured.
from taos's log I found that client is trying to connect to 26031 to execute create_db command

04/30 15:38:58.170746 00012690 RPC TSC 0x7fe7f485e188 0x4, expected create-db-rsp is not received
04/30 15:38:58.170964 00012690 RPC TSC 0x7fe7f485e188 0x4, failed to send msg:create-db to 192.168.1.231:26031
04/30 15:38:58.170998 00012690 RPC TSC 0x7fe7f485e188 0x4, rpc connection is released
04/30 15:38:58.175667 00012690 RPC TSC 0x4, connection error happens

I can not tell whether client use TCP or UDP to complete the rpc.

on the server side, I found that port 26031 is opened only with UDP protocol

tcp        0      0 0.0.0.0:26030           0.0.0.0:*               LISTEN      1/taosd             
tcp        0      0 0.0.0.0:26035           0.0.0.0:*               LISTEN      1/taosd             
tcp        0      0 0.0.0.0:26040           0.0.0.0:*               LISTEN      1/taosd             
tcp        0      0 0.0.0.0:26041           0.0.0.0:*               LISTEN      1/taosd             
tcp        0      0 172.20.3.73:26035       172.20.3.73:37420       ESTABLISHED 1/taosd             
tcp        0      0 172.20.3.73:37420       172.20.3.73:26035       ESTABLISHED 1/taosd             
udp        0      0 0.0.0.0:26030           0.0.0.0:*                           1/taosd             
udp        0      0 0.0.0.0:26031           0.0.0.0:*                           1/taosd             
udp        0      0 0.0.0.0:26035           0.0.0.0:*                           1/taosd

**Comment by zhanghongquan** [comment]
upload taoslog

**Comment by zhanghongquan** [comment]
[taoslog0.txt](https://github.com/taosdata/TDengine/files/6403842/taoslog0.txt)
upload taoslog

**Comment by zhanghongquan** [comment]
BTW it worked as expected while using taos shell

**Comment by yu285** [comment]
which OS you use in client side ? Linux?or windows? or you just run it on server side

**Comment by zhanghongquan** [comment]
I reproduce this problem using my work machine (windows 10, WSL).
After write a customized wireshark packet dissector and with the help of tcpdump, I found that this is because server (or docker-proxy indeed) did not forward the udp packet tothe right destination

![client](https://user-images.githubusercontent.com/3817333/117277299-56398300-ae92-11eb-9d9d-e0af430453e4.png)
![server](https://user-images.githubusercontent.com/3817333/117277304-576ab000-ae92-11eb-9c27-cdf8c6707bd3.png)
I attached two images here
in client.png, taos client send create_db command, but no response is gotten, in the meanwhile, at the server side, taos server did get a create_db command and responded it in less than 1 ms. but unfortunately, this udp packet is lost due to port unreachable.
I thought that this is because docker-proxy did not forward the udp packet to the right destination( client in trhis scenario).
I change the code to force to use tcp in rpc communication, and everything is OK.
I also create a pull request https://github.com/taosdata/TDengine/pull/6017 , add a new configuration to support rpcForceTcp.
